setup.py

```diff
@@ -32,13 +70,56 @@ def __init__(self, *p, **kw):
     unittest.runner.TextTestRunner = xmlrunner.XMLTestRunner
     del sys.argv[sys.argv.index('--xml')]
 
+setup_requires = ['Cython>=0.22']
+
 if '--no-cython' in sys.argv:
-    cythonize = None
     del sys.argv[sys.argv.index('--no-cython')]
 
-VERSION = "0.3.3"
+    # As hilarious as it sounds, when built with --no-cython, we DO
+    # need cython at runtime (to provide cython's shadow module), so
+    # remove it from setup_requires to keep it in install_requires
+    setup_requires = []
+elif cythonize is None:
+    from distutils.command.build import build
+    class MissingCython(build):
+        def run(self):
+            pkg_resources.require('Cython')
+    cmd_class.update({
+        'build' : MissingCython,
+        'install' : MissingCython,
```

Comment:
MissingCython only extends the build command. But on the install it will use the build. How does that works?